.github/workflows/codespell.yml

```diff
@@ -0,0 +1,23 @@
+# Codespell configuration is within setup.cfg
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```

Comment:
This will add an extra check on each PR, right?

Reply:
each PR against `main` but not other branches.  I could remove such limiter... but let's figure out pre-commit problem first.

Reply:
now action is gone in favor of pre-commit. I will "Resolve" this conversation